web/pdf_page_view.js

```diff
@@ -1036,19 +1036,22 @@ class PDFPageView {
     const sfx = approximateFraction(outputScale.sx);
     const sfy = approximateFraction(outputScale.sy);
 
-    canvas.width = floorToDivide(width * outputScale.sx, sfx[0]);
-    canvas.height = floorToDivide(height * outputScale.sy, sfy[0]);
-    const { style } = canvas;
-    style.width = floorToDivide(width, sfx[1]) + "px";
-    style.height = floorToDivide(height, sfy[1]) + "px";
+    const newWidth = (canvas.width = floorToDivide(
+      width * outputScale.sx,
+      sfx[0]
+    ));
+    const newHeight = (canvas.height = floorToDivide(
+      height * outputScale.sy,
+      sfy[0]
+    ));
+    const scaleX = newWidth / width;
+    const scaleY = newHeight / height;
```

Comment:
Unfortunately this seems wrong, since the condition below will no longer test the *correct* thing now.
```suggestion
    outputScale.sx = newWidth / width;
    outputScale.sy = newHeight / height;
```